sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -89,6 +90,8 @@ abstract class Optimizer(sessionCatalog: SessionCatalog, conf: CatalystConf)
       CombineFilters,
       CombineLimits,
       CombineUnions,
+      // Pushdown Filters again after combination
```

Comment:
#14912 is changed. But the changed solution is more complicated so I don't want to include it here. So I keep this change and wait for #14912 to be merged first.